### PR TITLE
feat(OMN-9763): add normalize_io_model_ref — dict-shape to dotted-string

### PR DESCRIPTION
**Pull request by jonahgabriel**
## Summary

Phase 2 / Task 6 of OMN-9757 (corpus classification + migration normalization layer).

Pure `dict -> dict` transform converting legacy `input_model: {name: X, module: Y}` shape to the canonical dotted string `"Y.X"` form expected by the strict typed contract models (`extra="forbid"`).

- Non-mutating, idempotent
- String passthrough preserved
- Missing field never injected
- Bootstraps `omnibase_core/normalization/` package + `tests/unit/normalization/` for sibling Phase 2 tasks (OMN-9761/9762/9764/9765/9766)

Parent epic: OMN-9757
Plan: docs/plans/2026-04-25-corpus-classification-and-normalization-layer.md (Task 6)

## Test plan

- [x] 5 unit tests pass (`uv run pytest tests/unit/normalization/test_contract_normalizer.py -v`)
- [x] mypy --strict clean (`uv run mypy src/omnibase_core/normalization/ tests/unit/normalization/ --strict`)
- [x] ruff format + check clean
- [x] pre-commit run --all-files passes
- [x] tests/unit/normalization/ + tests/unit/contracts/ — 793 passed locally
- [ ] Full CI green (gh pr checks --watch)

## Summary by CodeRabbit

* **New Features**
  * Contract normalization now standardizes input/output model references, converting structured refs into consistent dotted notation and leaving unsupported or missing values unchanged.

* **Tests**
  * New tests validate normalization behavior, edge cases, immutability, and idempotence to ensure reliable, predictable transformations.

### CODE REVIEW

File: src/omnibase_core/normalization/contract_normalizer.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+
+def _normalize_single_io_ref(value: dict[str, object] | str | None) -> str | None:
+    if value is None or isinstance(value, str):
+        return value
+    module = str(value.get("module", "") or "")
+    name = str(value.get("name", "") or "")
+    if module and name:
+        return f"{module}.{name}"
+    return module or name or None
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnibase_core/normalization/contract_normalizer.py]: Fixed in 34db45ed. Adopted the proposed fix verbatim —  now returns None unless both  and  are non-empty strings. The caller in  already treats None as 'leave untouched', so incomplete dicts are preserved for strict validation downstream. Added 4 new unit tests asserting this contract: module-only, name-only, empty-string field, and non-string field all pass through unchanged.
